specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,259 @@
+  - A game is a **latest anchor game** if it had the highest l2BlockNumber from the set of valid games known by this contract. It must be a valid game at the time it it set, but can be an invalid game in one specific case.
+- **Latest valid game**
+  - If the **latest anchor game** is a **valid game**, it is the **latest valid game**. Otherwise, there is no **latest valid game**.
+- **Dispute game finality delay**
+  - The dispute game finality delay is an **authorized input** representing the period of time between a dispute game
+    resolving and a dispute game becoming finalized or valid.
+  - Also known as "air gap."
+
+## Top-Level Invariants
+
+- The contract will only assert **valid games** are valid.
+- The latest anchor game must never serve the output root of a blacklisted game.
+- The latest anchor game must be recent enough so that the game doesn't break (run out of memory) in op-challenger.
+- The validity timestamp must start at zero.
+
+### Contract Dependents
+
+This contract manages and exposes dispute game validity so that other contracts can do things like correctly initialize
+dispute games and validate withdrawals.
+
+#### FaultDisputeGame
+
+A [FaultDisputeGame](fault-dispute-game.md) depends on this contract for a **latest valid anchor state** against which
+to resolve a claim and assumes its correct. Additionally, becauase proposers must gather L1 data for the window between
+the anchor state and the claimed state, FaultDisputeGames depend on this contract to keep a **latest valid anchor
+state** that's recent, so that proposer software is not overburdened (i.e. runs out of memory).
+
+#### OptimismPortal
+
+OptimismPortal depends on this contract to correctly report game validity as the basis for proving and finalizing
+withdrawals.
+
+- Can this dispute game can be used to prove a withdrawal? (Is the dispute game a **maybe valid game**?)
+- Can this dispute game can be used to finalize a withdrawal? (Is the dispute game a **valid game**?)
+
+### Contract Dependencies
+
+#### FaultDisputeGame
+
+Depends on FaultDisputeGame to correctly report:
+
+- whether its game type was the respected game type when created (and that it never changes once set).
+- its game type.
+- its l2BlockNumber.
+- its createdAt timestamp.
+- its resolvedAt timestamp.
+
+#### DisputeGameFactory
+
+Depends on DisputeGameFactory to correctly report:
+
+- whether a game was created by the DisputeGameFactory (is "factory-registered").
+
+#### SuperchainConfig
+
+Depends on SuperchainConfig to correctly report:
+
+- its guardian address.
+
+## Function-Level Invariants
+
+### `initialize`
+
+- Initial anchor state must be an **authorized input**.
+- Dispute game factory must be an **authorized input**.
+- `dispute game finality delay` must be an **authorized input**.
+- Superchain config must be an **authorized input**.
+
+### `getLatestValidGame`
+
+Returns **latest valid game**, or reverts if there is no **latest valid game**.
```

Comment:
Returns the **anchor game** if the **anchor game** is a **valid game**. Reverts if the **anchor game** is an **invalid game**